hscontrol/oidc.go

```diff
@@ -443,7 +443,9 @@ func (a *AuthProviderOIDC) createOrUpdateUserFromClaim(
 
 	// This check is for legacy, if the user cannot be found by the OIDC identifier
 	// look it up by username. This should only be needed once.
-	if user == nil {
+	// This branch will presist for a number of versions after the OIDC migration and
+	// then be removed following a deprecation.
+	if a.cfg.MapLegacyUsers && user == nil {
 		user, err = a.db.GetUserByName(claims.Username)
```

Comment:
I thought about this some more, and I don't think this migration is reliable either, because `preferred_username` might not be the same as the "user" part of an `email` address.

See my comments on `CHANGELOG.md`.